File: app/src/main/java/com/skybox/seven/covid/ui/AuthActivity.java
package com.skybox.seven.covid.ui;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

import com.akexorcist.localizationactivity.ui.LocalizationActivity;
import com.skybox.seven.covid.R;
import com.skybox.seven.covid.ui.fragment.LoginFragment;
import com.skybox.seven.covid.ui.fragment.RegisterFragment;


public class AuthActivity extends LocalizationActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_auth);
        Bundle extras = getIntent().getExtras();

        if (extras != null) {
            boolean register = extras.getBoolean("register");
            if (register) {
                RegisterFragment registerFragment = new RegisterFragment();
                FragmentManager fm = getSupportFragmentManager();

                fm.beginTransaction().add(R.id.mainlayout, registerFragment).commit();
            } else {
                LoginFragment loginFragment = new LoginFragment();
                FragmentManager fm = getSupportFragmentManager();

                fm.beginTransaction().add(R.id.mainlayout, loginFragment).commit();
            }
        } else {
            LoginFragment loginFragment = new LoginFragment();
            FragmentManager fm = getSupportFragmentManager();

            fm.beginTransaction().add(R.id.mainlayout, loginFragment).commit();
        }
    }
}

File: app/src/main/java/com/skybox/seven/covid/ui/AuthActivity.kt
package com.skybox.seven.covid.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.fragment.NavHostFragment
import com.skybox.seven.covid.R
import com.skybox.seven.covid.databinding.ActivityAuthBinding
import kotlinx.android.synthetic.main.activity_auth.*

class AuthActivity : AppCompatActivity() {
    lateinit var binding: ActivityAuthBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAuthBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val destination = if (intent.getBooleanExtra(LOGIN, true)) R.id.registerFragment else R.id.loginFragment
        val fragment = nav_host_fragment as NavHostFragment
        val graphInflater = fragment.navController.navInflater
        val navGraph = graphInflater.inflate(R.navigation.auth_navigation)

        navGraph.startDestination = destination
        fragment.navController.graph = navGraph

        binding.close.setOnClickListener {
            finish()
        }
    }

    companion object {
        const val LOGIN = "AUTH_LOGIN"
    }
}